scripts/RunHPCNodes.py

```diff
@@ -0,0 +1,115 @@
+# Taco Niet
+# Date:  September 26, 2016 - 2018
+# Script to run OSeMOSYS with various parameters using full nodes on a SLURM scheduler managed HPC system.
+
+#!/usr/bin/python
+import subprocess # for running glpsol
+from shutil import copy2 # Allow us to make the needed copies of the parameter file
+from math import sqrt
+import zipfile, os, time, io
+
+if __name__ == '__main__':
+    modelruns = []
+    tailcommands = []
+
+	# Setup task list here.  This can be created by a series of nested for loops.
+    param3List = ['10.6', '15.9', '21.2']
+    param2List = ['0000', '0050', '0100']
+    param1List = ['35', '65', '100']    
+
+    NumRunsTotal = len(param2s)*len(param3Levels)*len(param1s)
+    
+    NumRunstoGroup = 600  # Number of runs to group for each qsub command
+    NumberofCores = 32
+    WallTime = '06:00:00' # Walltime in hh:mm:ss format
+    ZipWallTime = '08:00:00' # Walltime in hh:mm:ss format for zipping files
+    
+    for param1 in param1List:
+        for param2 in param2List:
+            for param3 in param3List:
+                print('Setting up '+param1+' '+ param2+' '+param3+'.')
+                copy2('params_base.dat', param3 +'_'+ param1 +'_'+ param2 + 'params.dat')
+                f = open(param3 +'_'+ param1 +'_'+ param2 + 'params.dat', 'a') # Open file for appending...
+                f.write('\n')
+
+                # Write Parameter 1 into data file...
+                f.write('param	Parameter1 default 0	[REG,*,2020]	:=\n')
+                f.write('tparam3  '+str(float(param1)/1000)+'\n')
+                f.write('tEXAMPLE	1.000   ;\n\n')
+
+                # Set the param2 size:
+                f.write('param TechnologyMaxCapacity default 99999999999 := REG DAM 2020 '+param2+';\n\n')
+                
+                # Set the param3 residual capacity:
+                f.write('param	ResidualCapacity	default	0	:=[REG,*,*]	:   2020	:=\n')
+                f.write('tEXAMPLE	0.24    tparam3	'+param3+';\n\n')               
+               
+                # Specify output files
+                # NOTE:  This needs to be adjusted based on which version of OSeMOSYS you are using...  See  OutputFileAsParameter branch for one example.
+                f.write('param SelectedResults := \''+ param3 +'_'+ param1 +'_'+ param2 + 'SelectedResults.csv\';\n')
+                                    
+                # End the file and close it.
+                f.write('end;\n')
+                f.close()
+                
+                # append file name to list of files to process
+
+                modelruns.append('glpsol -m OSeMOSYS.mod -d ' + param3 +'_'+ param1 +'_'+ param2 + '_'+ 'params.dat --log '  + param3 +'_'+ param1 +'_'+ param2 + 'GLPScreen.txt -o ' + param3 +'_'+ param1 +'_'+ param2 + 'GLPOutput.txt')
```

Comment:
Might be cleaner to use `enumerate(zip())` above, and then use the index of the sample for the file names?

Reply:
That would be a nice way to do this.  Again, I'll look at this next time I use the script.